### PR TITLE
feat: validate specPattern root level

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [UNIFY-891](https://cypress-io.atlassian.net/browse/UNIFY-891)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by tgriesser — APPROVED** [review]

File: npm/react/examples/snapshots/cypress.config.js [hunk 1]
```diff
@@ -3,16 +3,16 @@ module.exports = {
...
-    '**/__snapshots__/*',
-    '**/__image_snapshots__/*',
-  ],
   'env': {
     'cypress-plugin-snapshots': {
       'prettier': true,
     },
   },
   'component': {
+    'ignoreSpecPattern': [
```
Comment (lmiller1990) [inline npm/react/examples/snapshots/cypress.config.js:12]: I wonder if this should be the default for CT. @cowboy maybe?

I also think `baseUrl` should be disallowed top level, it's E2E only. We should defs go over the config changes w/ a fine toothed comb to make sure everything is :100:, but this PR is definitely a good start.

File: packages/data-context/test/unit/sources/ProjectDataSource.spec.ts [hunk 2]
```diff
@@ -114,3 +120,60 @@ describe('transformSpec', () => {
...
+    'component/App.cy.ts',
+    'component/App.cy.js',
+    'e2e/onboarding.spec.ts',
+    'e2e/onboarding.cy.ts',
+    'e2e/onboarding.cy.js',
+  ]
+
+  let ctx: DataContext
+
+  beforeEach(async () => {
```
Comment (lmiller1990) [inline packages/data-context/test/unit/sources/ProjectDataSource.spec.ts:138]: Cool, I wonder if we want a `createTestDataContext` method that does all this - I think a lot of tests might benefit from it.

File: packages/data-context/test/unit/sources/ProjectDataSource.spec.ts [hunk 3]
```diff
@@ -114,3 +120,60 @@ describe('transformSpec', () => {
...
+    })
+
+    await Promise.all(fixture.map((element) => ctx.fs.outputFile(path.join(temporary, element), '')))
+  })
+
+  afterEach(async () => {
+    await ctx.fs.remove(temporary)
+  })
+
+  it('find all the *.cy.{ts,js} excluding the e2e', async () => {
```
Comment (lmiller1990) [inline packages/data-context/test/unit/sources/ProjectDataSource.spec.ts:162]: Nice, we definitely needed these tests.

**Review by lmiller1990 — APPROVED** [review]
Nice job, looks good - just some general comments, I wonder if we want to do a full DX review prior to merging this, or do that later? Eg `baseUrl` might need validating, also it seems like `__snapshots__` should be a default `ignoreSpecPattern` for CT.

File: packages/data-context/test/unit/sources/ProjectDataSource.spec.ts (hunk 2 above)
Comment (estrada9166) [reply packages/data-context/test/unit/sources/ProjectDataSource.spec.ts:138]: Yeah! we may want to move to a util, that can be really useful so we can create more unit tests for different dataSources and dataActions

**Review by tgriesser — APPROVED** [review]

**Pull request merged**